### PR TITLE
Add zonal isolation blog

### PR DESCRIPTION
**Pull request by Shaddoll**
Add zonal isolation blog

### CODE REVIEW

**Review by abhishekj720 — APPROVED** [review]

File: src/blog/_posts/zonal-isolation-v1.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+Infrastructure-level isolation is quite challenging to implement as it requires significant changes to the core design of the Cadence server. Due to the complexity involved, support for this feature is not planned for the foreseeable future.
+
+As a result, the focus remains on achieving task-level zonal isolation outside the Cadence server, which offers a more practical and immediate way to improve system resilience. It provides the capability of ensuring that an unhealthy zone (i.e. bad deployment of workers) only affect a subset of workflows (started from a certain zone) rather than every workflow in a Cadence domain.
+
+
+## How Zonal Isolation Works in Cadence?
+
+### Architecture
+Here is what the architecture of a zonally isolated Cadence-based system looks like:
+![zonal isolation overview](/img/zone-isolation.png)
```
Comment (taylanisikdemir) [inline src/blog/_posts/zonal-isolation-v1.md:27]: image seems broken when markdown is visualized https://github.com/uber/Cadence-Docs/blob/58a26015af67cb744bb757fc2e4504676daf003e/src/blog/_posts/zonal-isolation-v1.md
Please double check this is the right path format to insert images

File: src/blog/_posts/zonal-isolation-v1.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+- **Uber's Approach:** Let Cadence SDK set the origin zone in the headers of the requests before sending the request to Cadence.
+- **Preferred Approach:** Get the origin zone of the requests from headers set by network infrastructure.
+- Determine the origin zone of the requests from the zone of the cadence-frontend instance receiving the request, if the network layer has already achieved zonal isolation.
+
+The 2nd approach is the ideal one, but Uber's network infrastructure doesn't provide such headers and the network layer is not ready for zonal isolation. As a result, we adopt the 1st approach. At Uber, we have internal libraries in Go and Java acting as wrappers around Cadence SDK injecting necessary configurations. These libraries have been updated to include the origin zone in the request headers using a header called isolation-group.
```
Comment (taylanisikdemir) [inline src/blog/_posts/zonal-isolation-v1.md:39]: for those folks using OS cadence, can you explain what exact headers in the request they need to set?

File: src/blog/_posts/zonal-isolation-v1.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+For domain-level drain, the draining status is stored in [`domains`](https://github.com/uber/cadence/blob/v1.2.13/schema/cassandra/cadence/schema.cql#L412) table. For cluster-level drain, the status is stored in [`cluster_config`](https://github.com/uber/cadence/blob/v1.2.13/schema/cassandra/cadence/schema.cql#L498) table.
+
+## How to enable Zonal Isolation?
+### Server Update
+In order to enable this feature, please upgrade Cadence server to [v1.2.1](https://github.com/uber/cadence/releases/tag/v1.2.1) or later.
+
+**NOTE:** If you're not using the provided main binary located in `cmd/server`, you must adopt this [middleware](https://github.com/uber/cadence/blob/v1.2.1/common/rpc/middleware.go#L188). Depends on the appoach to [determine the origin zone of requests](#determine-the-zone-of-a-workflow-and-workers), you can adopt this [middleware](https://github.com/uber/cadence/blob/v1.2.1/common/rpc/middleware.go#L229) or build your own middleware using [`partition.ContextWithConfig`](https://github.com/uber/cadence/blob/v1.2.1/common/partition/context.go#L42) function injecting origin zone into the context.
+
+This feature is controled by 2 dynamic config properties.
```
Comment (taylanisikdemir) [inline src/blog/_posts/zonal-isolation-v1.md:64]: typo: controlled

File: src/blog/_posts/zonal-isolation-v1.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+- `system.allIsolationGroups`: This property provides the list of available zones within a region.
+- `system.enableTasklistIsolation`: This property enables Zonal Isolation at domain level.
+It’s important to note that any update to the `system.allIsolationGroups` property requires a restart of the cadence-matching service for the changes to take effect. This ensures that the new zone configuration is properly loaded and applied. However, the `system.enableTasklistIsolation` property can be updated dynamically without requiring a restart, making it more flexible for enabling or disabling Zonal Isolation on a per-domain basis.
```
Comment (taylanisikdemir) [inline src/blog/_posts/zonal-isolation-v1.md:67]: > requiring a restart

@davidporter-id-au After your changes for auto discovery, will it require restarting matching to recognize changes to `system.allIsolationGroups` ?

**Review by taylanisikdemir — APPROVED** [review]

File: src/blog/_posts/zonal-isolation-v1.md (hunk 1 above)
Comment (Shaddoll) [reply src/blog/_posts/zonal-isolation-v1.md:27]: it needs to be compiled to be visualized, so it's not visible in github

File: src/blog/_posts/zonal-isolation-v1.md (hunk 4 above)
Comment (davidporter-id-au) [reply src/blog/_posts/zonal-isolation-v1.md:67]: no

**Pull request merged**